### PR TITLE
Include handling of HNDTYPE_REFCOUNTED when FEATURE_OBJCWRAPPERS is defined

### PR DESCRIPTION
**Pull request by elinor-fung**
- Include handling of ref-counted handles when `FEATURE_OBJCWRAPPERS` (or `FEATURE_COMWRAPPERS`) is defined
  - Doesn't do anything right now, just builds
- Some scoping/separation of `FEATURE_COMWRAPPERS` vs `FEATURE_COMINTEROP` ifdefs

cc @AaronRobinsonMSFT @jkoritzinsky

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
I think this is fine. What are the throughs on creating a new `FEATURE_REFCOUNTEDHANDLE`?

**Comment by elinor-fung** [comment]
I had considered a new define (as some of them were getting up to four different defines...) - not opposed.

I didn't try add a new define because I had been thinking of feature defines as basically that concept/feature not existing if it is not defined? Right now, it seems like the ref-counted handle concept always exists, and it is just the handling/interpretation of `HNDTYPE_REFCOUNTED` that is `ifdef`-ed all over the place.

**Comment by AaronRobinsonMSFT** [comment]
@elinor-fung Fair point. LGTM. @jkoritzinsky ?

File: src/coreclr/utilcode/util.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ namespace
         StackSString ssDllName;
         if ((wszDllPath == nullptr) || (wszDllPath[0] == W('\0')) || fIsDllPathPrefix)
         {
-#ifndef TARGET_UNIX
+#ifdef FEATURE_COMINTEROP
```
Comment (jkotas) [inline src/coreclr/utilcode/util.cpp:240]: Nit: This is unrelated to COM interop. This path is used to load profilers or unmanaged symbol readers. It happens to use COM, but not COM interop. If we were to build CoreCLR runtime without COM interop, we would still want this path to work.

**Review by jkoritzinsky — APPROVED** [review]

File: src/coreclr/utilcode/util.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/utilcode/util.cpp:240]: It may be nice to change `#ifndef TARGET_UNIX` to `#ifdef HOST_WINDOWS` here.

Comment (elinor-fung) [reply src/coreclr/utilcode/util.cpp:240]: Ah, I had switched this because `FindInprocServer32UsingCLSID` in `utilcode.h` is under `FEATURE_COMINTEROP`. I'll make them `HOST_WINDOWS`.
